src/Microsoft.OData.Core/JsonLight/ODataJsonLightBatchPayloadItemPropertiesCache.cs

```diff
@@ -267,7 +273,13 @@ private void ScanJsonProperties()
                                 while (this.jsonReader.NodeType != JsonNodeType.EndObject)
                                 {
                                     string headerName = this.jsonReader.ReadPropertyName();
-                                    string headerValue = this.jsonReader.ReadPrimitiveValue().ToString();
+                                    string headerValue = this.jsonReader.ReadPrimitiveValue()?.ToString();

```

Comment:
I think we handle the `null` value scenario elsewhere. What scenarios is the null coalescing operator handling? Did you add tests for such scenarios?

Reply:
Not handling the `NULL` values, when executing the json property read function (including the header properties), will result in an `ArgumentNullException` and return an HTTP 500 "internal server error" by the application.
I've identified and described this behavior in the underlying issue ticket.
It's easy to reproduce, so it would be nice, if you have a look at it by yourself (just locally revert the changes and execute the corresponding `ReadBatchRequestWithNullHeadersAsync` UnitTest to verify).
https://github.com/OData/odata.net/issues/2656